### ISSUE TITLE
Support Filter in bool compound query

### ISSUE THREAD

**Issue by dbulaja98** [lambdaworks/zio-elasticsearch#63]
### Description:

At the moment, we have only `must` and `should` elements in bool query. The `filter` clause must appear in the matching documents. Unlike queries specified under `must`, queries specified under the `filter` element have no effect on scoring - scores are returned as 0.

```
GET _search
{
  "query": {
    "bool": {
      "filter": {
        "term": {
          "status": "active"
        }
      }
    }
  }
}
```

**Comment by kevchuang** [comment]
I would love to take this one !

**Comment by drmarjanovic** [comment]
@kevchuang all yours! 🎉

**Comment by drmarjanovic** [comment]
Closed with #86 

@kevchuang, thanks! We are looking forward to more contributions! 🥳 🎉
